bootloader/bootloader.go

```diff
@@ -95,6 +95,11 @@ type RecoveryAwareBootloader interface {
 	SetRecoverySystemEnv(recoverySystemDir string, values map[string]string) error
 }
 
+type ExtractedRecoveryKernelImageBootloader interface {
+	RecoveryAwareBootloader
```

Comment:
this looks wrong, the two kinds are distinct, maybe the other one needs a different name

Reply:
Yeah, sorry for that, this was a misunderstanding of mine, this needs to be undone.

Reply:
Dropped this.

As for the other name, maybe we should go with something like `RecoveryEnvironmentBootloader` instead of the generic one?

Reply:
we should rename but probably in a follow up, the other option would be:

ByEnvRecoveryKernelBootloader